modules/reindex/src/test/java/org/elasticsearch/index/reindex/RethrottleTests.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.reindex;
+
+import org.elasticsearch.action.ListenableActionFuture;
+import org.elasticsearch.action.admin.cluster.node.tasks.list.ListTasksResponse;
+
+import static org.hamcrest.Matchers.hasSize;
+
+/**
+ * Tests that you can set requests_per_second over the Java API and that you can rethrottle running requests. There are REST tests for this
+ * too but this is the only place that tests running against multiple nodes so it is the only integration tests that checks for
+ * serialization.
+ */
+public class RethrottleTests extends ReindexTestCase {
+
+    public void testReindex() throws Exception {
+        testCase(reindex().source("test").destination("dest"), ReindexAction.NAME);
+    }
+
+    public void testUpdateByQuery() throws Exception {
+        testCase(updateByQuery().source("test"), UpdateByQueryAction.NAME);
+    }
+
+    private <
+                Response extends BulkIndexByScrollResponse
```

Comment:
Does this need the indicator? This doesn't return anything anyway so can't it just be `private void testCase(...)`?